### ISSUE TITLE
Question about updating netD

### ISSUE THREAD

**Issue by tangjilin** [junyanz/pytorch-CycleGAN-and-pix2pix#344]
In the function optimize_parameters（），when updating D， we set  requires_grad（netD） ==True.
However, when updating G, we set  requires_grad（netD） ==False. However, it means that we do not  calculate the gradient of netD(I understand).  But the gradient that we use to update netG is from netD. How do we backward and  update netG with setting  requires_grad（netD)==False ?
Thanks!

**Comment by himat** [comment]
@tangjilin what was the answer?

**Comment by junyanz** [comment]
netG can still get the gradients. See this PyTorch [blog](https://pytorch.org/docs/stable/notes/autograd.html) for more details. According to the blog, "This (requires_grad) is especially useful when you want to freeze part of your model, or you know in advance that you’re not going to use gradients w.r.t. some parameters."

**Comment by jilner** [comment]
@tangjilin I have the same question, how to explain this ? thank you
